renku/service/views/cache.py

```diff
@@ -0,0 +1,335 @@
+              schema:
+                type: array
+              required: true
+
+            - in: query
+              name: unpack_archive
+              description: Determines if renku should work with archive.
+              schema:
+                type: bool
+              required: false
+
+        responses:
+            result:
+                description: Contains `file_id` reference.
+
+            error:
+                description: Contains `code` and `message` \
+                 describing the error.
+
+    """
+    if request.method == 'POST':
+
+        if 'file' not in request.files:
+            return jsonify(
+                error={
+                    'code': INVALID_PARAMS_ERROR_CODE,
+                    'message': 'no file parts'
+                }
+            )
+
+        file = request.files['file']
+        if file.filename == '':
+            return jsonify(
+                error={
+                    'code': INVALID_PARAMS_ERROR_CODE,
+                    'message': 'no file selected for uploading'
+                }
+            )
+
+        if file and allowed_file(file.filename):
+            unpack_archive = request.args.get('unpack_archive', False)
+            filename = secure_filename(file.filename)
+            timestamp = time.time() * 1e+9
+
+            file_metadata = {
+                'filename': filename,
+                'timestamp': timestamp,
+                'user': user,
+                'unpack_archive': unpack_archive
+            }
+
+            encoded = jwt.encode(
+                file_metadata, JWT_KEY, algorithm=JWT_ALGORITHM
+            )
+
+            local_path = CACHE_UPLOADS_PATH / Path(encoded.decode('utf-8'))
+            file.save(str(local_path))
+            file_id = encoded.decode('utf-8').split('.')[1]
+
+            return jsonify(result={'file_id': file_id})
+
+        return jsonify(
+            error={
+                'code': INVALID_PARAMS_ERROR_CODE,
+                'message': 'file type not allowed'
+            }
+        )
+
+
+@requires_identity
+def clone_repository_view(user):
+    """
+    Clone remote repository.
+
+    ---
+    get:
+        tags:
+        - "cache"
+
+        summary: Clone remote repository.
+        description: Clone remote repository specified by git_url argument.
+
+        parameters:
+            - in: header
+              name: Authorization
+              schema:
+                type: string
+              required: true
+              description: User identification.
+
+            - in: query
+              name: git_url
+              description: Remote URL of git repository.
+              schema:
+                type: string
+              required: true
+
+            - in: query
+              name: git_username
+              description: Username to access remote git repository.
+              schema:
+                type: string
+              required: true
+
+            - in: query
```

Comment:
I don't think you want tokens in query parameters? It's going to be all over the logs. You could request the access token from the gateway?